components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1472,6 +1477,64 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
         }
     }
 
+    /**
+     * Updates the application role mapping configuration.
+     *
+     * @param serviceProvider ServiceProvider to be updated.
+     * @param applicationRoleMappingConfigs Application role mapping configurations.
+     * @param connection Connection to the database.
+     * @throws SQLException SQLException when updating application role mapping configuration.
+     */
+    private void updateAppRoleMappingConfiguration(ServiceProvider serviceProvider,
+                                                   AppRoleMappingConfig[] applicationRoleMappingConfigs,
+                                                   Connection connection)
+            throws SQLException {
+
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        int applicationId = serviceProvider.getApplicationID();
+        List<String> attributeStepFIdPs =
+                getAttributeStepFIdPs(serviceProvider.getLocalAndOutBoundAuthenticationConfig());
+        deleteAppRoleMappingConfiguration(applicationId, connection);
+
+        try (PreparedStatement updateAppRoleMappingPrepStmt = connection
+                .prepareStatement(ApplicationMgtDBQueries.UPDATE_SP_IDP_ATTR)) {
+            if (applicationRoleMappingConfigs != null) {
+                for (AppRoleMappingConfig applicationRoleMappingConfig : applicationRoleMappingConfigs) {
+                    String fidPName = applicationRoleMappingConfig.getIdPName();
+                    if (attributeStepFIdPs.contains(fidPName)) {
+                        // Get idp id using idp name.
+                        int idpId = getIdPId(connection, tenantID, applicationRoleMappingConfig.getIdPName());
+                        updateAppRoleMappingPrepStmt.setInt(1, applicationId);
+                        updateAppRoleMappingPrepStmt.setInt(2, idpId);
+                        updateAppRoleMappingPrepStmt.setString(3, APP_ROLE_MAPPINGS_KEY);
+                        updateAppRoleMappingPrepStmt.setString(4, applicationRoleMappingConfig
+                                .isUseAppRoleMappings() ? "1" : "0");
+                        updateAppRoleMappingPrepStmt.addBatch();
+                    }
+                }
+                updateAppRoleMappingPrepStmt.executeBatch();
+            }
+        }
+    }
+
+    /**
+     * Deletes the application role mapping configuration.
```

Comment:
```suggestion
     * Delete the application role mapping configuration.
```

Reply:
Addressed in 4cdf9672bdd6a4d6060995024d14bd9dc79a1168